### PR TITLE
merge remainder of 1.2.1 (ipy3 compatibility) to master

### LINKED ISSUE #1328: merge ipython3 fixes from v1.2 branch to master
finish https://github.com/twosigma/beaker-notebook/issues/1222

### PR DESCRIPTION
**Pull request by scottdraves**
also fixes up the moment situation https://github.com/twosigma/beaker-notebook/issues/1388
and closes https://github.com/twosigma/beaker-notebook/issues/1328

davide please double check the finalStuff/doFinish stuff , i am not sure how to repro that.

### CODE REVIEW

File: plugin/ipythonPlugins/src/dist/ipython/ipython.js [hunk 1]
```diff
@@ -235,26 +270,34 @@ define(function(require, exports, module) {
             } else {
               evaluation.jsonres = jsonres;
               var elem = $(document.createElement("div"));
-              var oa = new myPython.OutputArea(elem);
+	      var oa = (ipyVersion == '3') ?
+                (new myPython.OutputArea({events: {trigger: function(){}},
```
Comment (ildipo) [inline plugin/ipythonPlugins/src/dist/ipython/ipython.js:261]: This block here is redundant: it should only be done in the else branch of the following if.

File: plugin/ipythonPlugins/src/dist/iruby/iruby.js [hunk 2]
```diff
@@ -27,267 +27,341 @@ define(function(require, exports, module) {
...
+          } else if (type === "stream") {
+            evaluation.outputdata = [];
+            if (finalStuff !== undefined && finalStuff.outputdata !== undefined)
+              evaluation.outputdata = finalStuff.outputdata;
+            var text = (ipyVersion == '3') ? content.text : content.data;
+            evaluation.outputdata.push({type: (content.name === "stderr") ? 'err' : 'out',
+                                        value: text});
           } else {
-            evaluation.payload = elem.html();
+            var elem = $(document.createElement("div"));
```
Comment (ildipo) [inline plugin/ipythonPlugins/src/dist/iruby/iruby.js]: also this block here is redundant (from line 248 to 260): the same code is below in the else branch

File: plugin/ipythonPlugins/src/dist/julia/julia.js [hunk 3]
```diff
@@ -217,25 +241,52 @@ define(function(require, exports, module) {
...
               oa.append_mime_type(content.data, elem);
+            } else {
+              oa.append_mime_type(content, elem);
+            }
+            var jsonres;
+            if(content.data['application/json'] !== undefined) {
+              jsonres = JSON.parse(content.data['application/json']);
             }
-            var table = bkHelper.findTable(elem[0]);
-            if (table) {
```
Comment (ildipo) [inline plugin/ipythonPlugins/src/dist/julia/julia.js:235]: Same as for python and iruby

File: plugin/ipythonPlugins/src/dist/python3/python3.js [hunk 4]
```diff
@@ -217,12 +241,23 @@ define(function(require, exports, module) {
             evaluation.outputdata = [];
             if (finalStuff !== undefined && finalStuff.outputdata !== undefined)
               evaluation.outputdata = finalStuff.outputdata;
-            if (content.name === "stderr") {
-              evaluation.outputdata.push( { type : 'out', value : content.data } );
+            var text = (ipyVersion == '3') ? content.text : content.data;
+            evaluation.outputdata.push({type: (content.name === "stderr") ? 'err' : 'out',
+                                        value: text});
+          } else {
+            var elem = $(document.createElement("div"));
```
Comment (ildipo) [inline plugin/ipythonPlugins/src/dist/python3/python3.js]: and here too

**Comment by ildipo** [comment]
It does not work at all:
Failed to load resource: the server responded with a status of 404 (Not Found)
beaker.js:409 ReferenceError: requirejs is not defined

**Comment by samccone** [comment]
It looks like this was the first merge to master that started a trend of failing specs
https://travis-ci.org/twosigma/beaker-notebook/builds/57523288

Before this point master was green. 

It might be worth investigating why the specs went red. any thoughts @buob @scottdraves @ildipo ?

**Comment by scottdraves** [comment]
Thanks for looking into this, Sam.
I am a little confused though, the build that you link to and the PR you
comment on are different.
Looking at their contents surely it is this one #1393, and in particular
the change in location of require.js that's causing the problem?

On Tue, Apr 28, 2015 at 7:13 PM, Sam Saccone notifications@github.com
wrote:

> It looks like this was the first merge to master that started a trend of
> failing specs
> https://travis-ci.org/twosigma/beaker-notebook/builds/57523288
> 
> Before this point master was green.
> 
> It might be worth investigating why the specs went red. any thoughts @buob
> https://github.com/buob @scottdraves https://github.com/scottdraves
> @ildipo https://github.com/ildipo ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/twosigma/beaker-notebook/pull/1393#issuecomment-97258699
> .

## 

BeakerNotebook.com
ScottDraves.com http://www.ScottDraves.com
@Scott_Draves http://twitter.com/scott_draves

**Comment by samccone** [comment]
this PR contains the commit that caused master to start being red

build:
https://travis-ci.org/twosigma/beaker-notebook/builds/57358180

commit sha:
https://github.com/twosigma/beaker-notebook/commit/3da2e14d6e9634a1d34f5e20f8d06c1f7f862254

sorry I copied the incorrect build number from travis

it very well might be related to requirejs it looks as if nothing is loading. I have not dug in to the cause of the error, just trying to give us a decent starting place

**Pull request merged**